server/src/main/java/org/tctalent/server/repository/db/SavedSearchSpecification.java

```diff
@@ -64,84 +66,80 @@ public static Specification<SavedSearch> buildSearchQuery(
-            if (savedSearchType != null) {
-                // Search for specified type
-                SavedSearchSubtype savedSearchSubtype = request.getSavedSearchSubtype();
-                String type = SavedSearch.makeStringSavedSearchType(
-                        savedSearchType, savedSearchSubtype);
-                conjunction.getExpressions().add(
-                        builder.equal(savedSearch.get("type"), type)
-                );
-            }
-
-            //If fixed is specified, only supply matching saved searches
-            if (request.getFixed() != null && request.getFixed()) {
-                conjunction.getExpressions().add(
-                        builder.equal(savedSearch.get("fixed"), request.getFixed())
-                );
-            }
-
-            // (shared OR owned OR global)
-            Predicate ors = builder.disjunction();
-
-            if (request.getGlobal() != null && request.getGlobal()) {
-                ors.getExpressions().add(
-                        builder.equal(savedSearch.get("global"), request.getGlobal())
-                );
-            }
-
-            //If shared is specified, only supply searches shared with the owner
-            if (request.getShared() != null && request.getShared()) {
-                if (loggedInUser != null) {
-                    Set<SavedSearch> sharedSearches = loggedInUser.getSharedSearches();
-                    if (!sharedSearches.isEmpty()) {
-                        Set<Long> sharedIDs = new HashSet<>();
-                        for (SavedSearch sharedSearch : sharedSearches) {
-                            sharedIDs.add(sharedSearch.getId());
-                        }
-                        ors.getExpressions().add(
-                                savedSearch.get("id").in( sharedIDs )
-                        );
-                    }
-                }
-            }
-
-            //If owned by this user (ie by logged in user)
-            if (request.getOwned() != null && request.getOwned()) {
-                if (loggedInUser != null) {
-                    ors.getExpressions().add(
-                         builder.equal(savedSearch.get("createdBy"), loggedInUser.getId())
-                    );
-                }
-            }
-
-            if (ors.getExpressions().size() != 0) {
-                conjunction.getExpressions().add(ors);
-            }
-
-            return conjunction;
-        };
-    }
-
-}
+      // ONLY SHOW ACTIVE SAVED SEARCHES
+      conjunction.getExpressions().add(
+          builder.equal(savedSearch.get("status"), Status.active)
+      );
+
+      // DON'T SHOW DEFAULT SAVED SEARCHES
+      conjunction.getExpressions().add(
+          builder.not(savedSearch.get("defaultSearch"))
+      );
+
+      // Filter by keyword if present
+      if (!StringUtils.isBlank(request.getKeyword())) {
+        String lowerCaseMatchTerm = request.getKeyword().toLowerCase();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### toLowerCase [](|2045d6a2-68fa-4cde-87a6-566ffddb2b76|)

The parameterless versions of `String.toLowerCase()` and `String.toUpperCase()` use the default locale of the JVM when transforming strings. This can have unintended results, including difficult-to-debug transient failures because case mapping differs based on locale. 

**Even if you aren’t writing code for a user experience that’s displayed to a customer**, system internal code often uses case folding to normalize strings for comparison.
Not doing so can missing locale lead to errors when running on systems that are configured to use a different system locale. This can include developer desktops or build servers (not just your fleet hosts): many hours have been spent debugging errors that only occur on a specific machine due to its locale configuration.

#### Recommended solutions:

Always pass in a `Locale`.

**Language or locale-dependent processing:**

Get the locale from the platform, use the language of the content, or configure it in your code and pass that to the method.
```
String firstName = &#34;Ichabod&#34;; 
Locale locale = // Where you obtain the locale from is platform dependent
String lowerCaseFirstName = firstName.toLowerCase(locale);
``` 

**Internal processing that isn’t locale-dependent:**

*Example: S3 bucket name*

Always specify the language/country-neutral locale `Locale.ROOT`. 

```
String imgTag = &#34;IMG&#34;; 
if (&#34;img&#34;.equals(imgTag.toLowerCase(Locale.ROOT)) {    
// do something 
}
```